### PR TITLE
feat: check user email from gravity instead of args

### PR DESCRIPTION
**Pull request by AlicanAkyuz**
### Description 

This PR improves on security concerns with regards to adding a user to anonymously submitted submissions by: 
- checking the match between the email address on the submission and user's gravity record

### CODE REVIEW

File: app/graphql/resolvers/base_resolver.rb [hunk 1]
```diff
@@ -42,8 +42,9 @@ def matching_user(submission, session_id)
       submitted_by_current_session?(submission, session_id)
   end
 
-  def matching_email(submission, user_email)
-    submission.user_email.downcase == user_email.downcase
+  def matching_email(submission)
+    user = Gravity.client.user(id: @context[:current_user]).user_detail._get
```
Comment (kiryl-zubarau) [inline app/graphql/resolvers/base_resolver.rb:46]: Could you please add try...catch here

Comment (AlicanAkyuz) [reply app/graphql/resolvers/base_resolver.rb:46]: I think line 11 in `add_user_to_submission_resolver` would catch it if anything goes wrong here. but I'll add a more specific exception handler 👍

**Review by kiryl-zubarau — APPROVED** [review]

File: app/graphql/resolvers/base_resolver.rb [hunk 2]
```diff
@@ -42,8 +42,14 @@ def matching_user(submission, session_id)
...
   end
 
-  def matching_email(submission, user_email)
-    submission.user_email.downcase == user_email.downcase
+  def matching_email(submission)
+    begin
+      user = Gravity.client.user(id: @context[:current_user]).user_detail._get
+      submission.user_email.downcase == user.email
+    rescue StandardError
+      Rails.logger.info 'Unable to match user email with submission email'
```
Comment (iskounen) [inline app/graphql/resolvers/base_resolver.rb:50]: ❤️

File: app/graphql/resolvers/base_resolver.rb [hunk 3]
```diff
@@ -42,8 +42,14 @@ def matching_user(submission, session_id)
       submitted_by_current_session?(submission, session_id)
   end
 
-  def matching_email(submission, user_email)
-    submission.user_email.downcase == user_email.downcase
+  def matching_email(submission)
+    begin
+      user = Gravity.client.user(id: @context[:current_user]).user_detail._get
+      submission.user_email.downcase == user.email
```
Comment (iskounen) [inline app/graphql/resolvers/base_resolver.rb:48]: **suggestion:** @AlicanAkyuz should we also `.downcase` the user.email?

**Pull request merged**